src/parallel/parallel_ghost_sync.C

```diff
@@ -88,4 +88,32 @@ void SyncSubdomainIds::act_on_data (const std::vector<dof_id_type> & ids,
     }
 }
 
+SyncElementIntegers::SyncElementIntegers(MeshBase & m, const std::string & integer_name)
+  : mesh(m), ind(m.get_elem_integer_index(integer_name))
+{}
+
+void SyncElementIntegers::gather_data(const std::vector<dof_id_type> & ids,
+                                      std::vector<dof_id_type> & ids_out) const
+{
+  ids_out.reserve(ids.size());
+
+  for (const auto & id : ids)
+    {
+      Elem * elem = mesh.query_elem_ptr(id);
```

Comment:
Likewise here.  See the nearby SyncSubdomainIds functor for example.  If you want to demand that you always find the element you're querying, we have a shorter and less indirect API for that.